### ISSUE TITLE
app crashes when switching models

### ISSUE THREAD

**Issue by IlianP** [Saik0s/Whisperboard#8]
tested on an iPhone 8 with iOS 16.3.1

The first model I had (successfully) downloaded was "ggml-small.bin". After testing it (see my other issue: #7 ), then I downloaded the model "ggml-tiny.bin" but now, every time I try to select it in the settings, the app completely crashes (without giving any error message). It takes me back to iOS homescreen right away.

Haven't tried a complete re-install yet (which will most probably solve this).

**Comment by Saik0s** [comment]
Hi @IlianP, thanks for reporting this. Does it happen on the currently available AppStore version?

**Comment by IlianP** [comment]
I had installed it from the Apple app store yesterday, so I suppose it was/still is the newest version (since there was no update since yesterday).

Now I just deleted in installed it again and I wasn't able to reproduce the same crash anymore. Will try to reproduce again with another combination/order of downloaded models after a new installation.

**Comment by IlianP** [comment]
So I tried different test combinations with a fresh install where I first downloaded a tiny, base or small model. Then downloading (at least) another model next. The result was: It doesn't matter whether I downloaded the tiny, base or small model first; it **always** crashes when trying to switch:
- from a tiny or base model to a small model
- from a small model to a tiny or base model

It does **not** crash when switching between any tiny/base models. So for my setup, the (biggest) small models seem to be the issue.
